src/chrome/content/rules/Wolfram_Alpha.xml

```diff
@@ -1,34 +1,65 @@
 <!--
-Invalid certificate:
-	api-cn.wolframalpha.com
-	api-maps.wolframalpha.com
-	api-tw.wolframalpha.com
-	dev1.cvs.wolframalpha.com
-	dev2.cvs.wolframalpha.com
-	reg.cvs.wolframalpha.com
-	sit.cvs.wolframalpha.com
-	cvsrx.wolframalpha.com
-	cvsrx.devel.wolframalpha.com
-	www-cn.wolframalpha.com
-	www-tw.wolframalpha.com
-	volunteer.wolframalpha.com
-
-Refused:
-	community.wolframalpha.com
-
-Timeout:
-	blog.wolframalpha.com
-	nba.devel.wolframalpha.com
-	homeworkday.wolframalpha.com
-	www2.wolframalpha.com
+	For other Wolfram Research coverage:
+
+		- Wolfram.com.xml
+		- Wolfram_CDN.com.xml
+		- Wolframcloud.com.xml
+
+	Problematic hosts in *wolframalpha.com:
+
+		- api-cn (http only)
+		- api-maps (http only)
+		- api-tw (http only)
+		- blog (http only)
+		- citi (http only)
+		- cvsrx (sha1)
+		- cvsrx.devel (mismatched)
+		- community (http only)
+		- homeworkday (timeout)
+		- morganstanley (untrusted)
+		- nielsen (sha1)
+		- volunteer (sha1)
+		- www-cn (sha1)
+		- www-tw (sha1)
+		- www[0-9][0-9] (http only)
+
 -->
 <ruleset name="Wolfram Alpha">
+
+	<target host="wolframalpha.com" />
+	<target host="api.wolframalpha.com" />
+	<target host="api-cn.wolframalpha.com" />
+	<target host="api-maps.wolframalpha.com" />
+	<target host="api-tw.wolframalpha.com" />
+	<target host="developer.wolframalpha.com" />
+	<target host="m.wolframalpha.com" />
+	<target host="preview.wolframalpha.com" />
+	<target host="products.wolframalpha.com" />
+	<target host="volunteer.wolframalpha.com" />
+	<target host="wc.wolframalpha.com" />
+	<target host="www1.wolframalpha.com" />
+	<target host="www3.wolframalpha.com" />
+	<target host="www4b.wolframalpha.com" />
+	<target host="www4c.wolframalpha.com" />
+	<target host="www4d.wolframalpha.com" />
+	<target host="www4f.wolframalpha.com" />
+	<target host="www5a.wolframalpha.com" />
+	<target host="www5b.wolframalpha.com" />
```

Comment:
These targets are redundant to the wildcard. Please transform them into test urls.

Reply:
The wildcard regex only covers `^http://(www[0-9][0-9]\.)?wolframalpha\.com/`. Wolfram Alpha does not have HTTPS support on all subdomains, and hence replacing these with test urls would require the list to also be compacted into the regex, a practice that to my understanding is explicitly discouraged (?). It's not an ideal overlap, but this PR cuts the number of Wolfram rulesets in the whitelist from 3 to 1.

Reply:
This ruleset contains `<target host="*.wolframalpha.com" />` and `<rule from="^http:" to="https:" />` so it will rewrite anything to https. But you also listed some explicit targets like `www5b.wolframalpha.com` and special rewrites.
This is confusing and I think it does not behave like you planned.

So I would list all targets, keep the generic redirect and remove the wildcard target, or use a wildcard target with specific redirect rules and test urls.

Reply:
> I think it does not behave like you planned.

You are very correct, such a rule would redirect all subdomains to HTTPS indiscriminately. Scarily, that seems to have also been the behavior of the previous rule, and probably explains why I copied it without thinking.